lib/internal/modules/esm/translators.js

```diff
@@ -86,32 +97,90 @@ translators.set('module', async function moduleStrategy(url) {
   return module;
 });
 
+function parseCJSModuleExports(filename) {
+  const cached = processLoader.ESMLoader.cjsCache.get(filename);
+  if (cached)
+    return cached.exportNames;
+
+  const source = readFileSync(filename, 'utf8');
```

Comment:
I'm not a big fan of two different files both having independent implementations of the same logic. Can we add comments on each of the `readFileSync` lines warning that a change to one has to be applied to the other as well? Or, alternatively, move the `cjsCache` to the CommonJS loader. E.g. `CJSLoader.prefetchSource(filename)` here and have CJSLoader own reading from and writing to the cache (including analyzing the CJS source).